external/avro/src/main/scala/org/apache/spark/sql/avro/AvroUtils.scala

```diff
@@ -228,6 +236,38 @@ private[sql] object AvroUtils extends Logging {
       .groupBy(_.name.toLowerCase(Locale.ROOT))
       .mapValues(_.toSeq) // toSeq needed for scala 2.13
 
+    private[this] val fieldPairs = catalystSchema
+      .zipWithIndex
+      .map(fieldWithPos => (fieldWithPos, getAvroField(fieldWithPos._1.name, fieldWithPos._2)))
+
+    /**
+     * Get the fields which have matching equivalents in both Avro and Catalyst schemas.
+     *
+     * @return A sequence of `(catalystField, catalystPosition, avroField)` tuples.
+     */
+    def getMatchedFields: Seq[(StructField, Int, Schema.Field)] = fieldPairs.flatMap {
```

Comment:
The result of this method is complicated.

Reply:
Let's create a new case class for this:
```
case class AvroMatchedField(avroField Schema.Field, catalystField StructField, catalystPosition int)
```

Reply:
Great suggestion, thanks!